lib/web/assistant.go

```diff
@@ -710,13 +673,30 @@ func (h *Handler) assistChatLoop(ctx context.Context, assistClient *assist.Assis
 
 		// Token usage reporting is asynchronous as we might still be streaming
 		// a message, and we don't want to block everything.
-		go h.reportTokenUsage(usedTokens, lookaheadTokens, conversationID, authClient)
+		go h.reportTokenUsage(usedTokens, conversationID, authClient)
 	}
 
 	h.log.Debug("end assistant conversation loop")
 	return nil
 }
 
+// preliminaryRatelimitGuard checks that some small amount of tokens are still available and the ratelimit is not exceeded.
+// This is done because the changed quantity within the limiter is not known until after a request is processed.
```

Comment:
Can we update the godocs to mention what `onMessageFn` is and under what circumstances it is called?